packages/core/src/lighting/screenSpaceLighting/ScreenSpaceAmbientOcclusionEffect.ts

```diff
@@ -0,0 +1,178 @@
+import { Shader, ShaderMacro, ShaderPass, ShaderProperty } from "../../shader";
+import blitVs from "../../shaderlib/extra/Blit.vs.glsl";
+import scalableAmbientOcclusionFS from "./shaders/ScalableAmbientOcclusion.glsl";
+import bilateralBlurFS from "./shaders/Blur/BilateralBlur.glsl";
+
+/**
+ * Screen Space Ambient Occlusion quality levels.
+ */
+export enum SSAOQuality {
+  /** Low quality - fewer samples, faster performance */
+  Low = 0,
+  /** Medium quality - balanced samples and performance */
+  Medium = 1,
+  /** High quality - more samples, better quality */
+  High = 2
+}
+
+/**
+ * Screen Space Ambient Occlusion effect configuration.
+ */
+export class ScreenSpaceAmbientOcclusion {
+  static readonly SHADER_NAME = "ScalableAmbientOcclusion";
+
+  // Shader properties for ambient occlusion calculation
+  /** @internal */
+  static _invRadiusSquaredProp = ShaderProperty.getByName("material_invRadiusSquared");
+  /** @internal */
+  static _intensityProp = ShaderProperty.getByName("material_intensity");
+  /** @internal */
+  static _projectionScaleRadiusProp = ShaderProperty.getByName("material_projectionScaleRadius");
+  /** @internal */
+  static _biasProp = ShaderProperty.getByName("material_bias");
+  /** @internal */
+  static _minHorizonAngleSineSquaredProp = ShaderProperty.getByName("material_minHorizonAngleSineSquared");
+  /** @internal */
+  static _peak2Prop = ShaderProperty.getByName("material_peak2");
+  /** @internal */
+  static _powerProp = ShaderProperty.getByName("material_power");
+  /** @internal */
+  static _invPositionProp = ShaderProperty.getByName("material_invProjScaleXY");
+
+  // Shader properties for bilateral blur
+  /** @internal */
+  static _farPlaneOverEdgeDistanceProp = ShaderProperty.getByName("material_farPlaneOverEdgeDistance");
+  /** @internal */
+  static _kernelProp = ShaderProperty.getByName("material_kernel");
+
+  // Shader macros
+  /** @internal */
+  static _enableMacro = ShaderMacro.getByName("ENABLE_SSAO");
```

Comment:
All naming must follow the engine naming conventions，such as `CAMERA_ENABLE_***`, `material_Bias`